src/Operators/finitedifference.jl

```diff
@@ -1592,6 +1592,214 @@ end
+    Rⱼ₊₁⁻ = (Pⱼ₊₁⁻ > 0 ? min(1, Qⱼ₊₁⁻ / Pⱼ₊₁⁻) : 0)
+
+    Cⱼ₊₁₂ = (Aⱼ₊₁₂ ≥ 0 ? min(Rⱼ₊₁⁺, Rⱼ⁻) : min(Rⱼ⁺, Rⱼ₊₁⁻))
+
+    return Cⱼ₊₁₂ * Aⱼ₊₁₂
+
+end
+
+stencil_interior_width(::FCTZalesak, A_space, Φ_space, Φᵗᵈ_space) =
+    ((0, 0), (-half - 1, half + 1), (-half - 1, half + 1))
+
+@inline function stencil_interior(
+    ::FCTZalesak,
+    loc,
+    idx,
+    hidx,
+    A_space,
+    Φ_space,
+    Φᵗᵈ_space,
+)
+
+    center_space = axes(Φ_space)
+    face_space = axes(A_space)
+    # cell center variables
+    ϕⱼ₋₁ = getidx(Φ_space, loc, idx - half - 1, hidx)
+    ϕⱼ = getidx(Φ_space, loc, idx - half, hidx)
+    ϕⱼ₊₁ = getidx(Φ_space, loc, idx + half, hidx)
+    ϕⱼ₊₂ = getidx(Φ_space, loc, idx + half + 1, hidx)
+    # cell center variables
+    ϕⱼ₋₁ᵗᵈ = getidx(Φᵗᵈ_space, loc, idx - half - 1, hidx)
+    ϕⱼᵗᵈ = getidx(Φᵗᵈ_space, loc, idx - half, hidx)
+    ϕⱼ₊₁ᵗᵈ = getidx(Φᵗᵈ_space, loc, idx + half, hidx)
+    ϕⱼ₊₂ᵗᵈ = getidx(Φᵗᵈ_space, loc, idx + half + 1, hidx)
+    # cell face variables
+    Aⱼ₊₁₂ = Geometry.contravariant3(
+        getidx(A_space, loc, idx, hidx),
+        Geometry.LocalGeometry(face_space, idx, hidx),
+    )
+    Aⱼ₋₁₂ = Geometry.contravariant3(
+        getidx(A_space, loc, idx - 1, hidx),
+        Geometry.LocalGeometry(face_space, idx - 1, hidx),
+    )
+    Aⱼ₊₃₂ = Geometry.contravariant3(
+        getidx(A_space, loc, idx + 1, hidx),
+        Geometry.LocalGeometry(face_space, idx + 1, hidx),
+    )
+
+
+    dxⱼ = Geometry.LocalGeometry(center_space, idx - half, hidx).∂x∂ξ[1]
+    dxⱼ₊₁ = Geometry.LocalGeometry(center_space, idx + half, hidx).∂x∂ξ[1]
+
+
+    return Geometry.Contravariant3Vector(
+        fct_zalesak(
+            Aⱼ₋₁₂,
+            Aⱼ₊₁₂,
+            Aⱼ₊₃₂,
+            ϕⱼ₋₁,
+            ϕⱼ,
+            ϕⱼ₊₁,
+            ϕⱼ₊₂,
+            ϕⱼ₋₁ᵗᵈ,
+            ϕⱼᵗᵈ,
+            ϕⱼ₊₁ᵗᵈ,
+            ϕⱼ₊₂ᵗᵈ,
+            dxⱼ,
+            dxⱼ₊₁,
+        ),
+    )
+end
+
+boundary_width(
+    ::FCTZalesak,
+    ::ThirdOrderOneSided,
+    A_space,
+    Φ_space,
+    Φᵗᵈ_space,
+) = 2
+
+@inline function stencil_left_boundary(
+    ::FCTZalesak,
+    bc::ThirdOrderOneSided,
+    loc,
+    idx,
+    hidx,
+    A_space,
+    Φ_space,
+    Φᵗᵈ_space,
+)
+    face_space = axes(A_space)
+    @assert idx <= left_face_boundary_idx(face_space) + 1
+
+    Aⱼ₊₁₂ = Geometry.contravariant3(
+        getidx(A_space, loc, idx, hidx),
+        Geometry.LocalGeometry(face_space, idx, hidx),
+    )
+    return Geometry.Contravariant3Vector(Aⱼ₊₁₂ * 0)
+end
+
+@inline function stencil_right_boundary(
+    ::FCTZalesak,
+    bc::ThirdOrderOneSided,
+    loc,
+    idx,
+    hidx,
+    A_space,
+    Φ_space,
+    Φᵗᵈ_space,
+)
+    face_space = axes(A_space)
+    @assert idx <= right_face_boundary_idx(face_space) - 1
+
+    Aⱼ₊₁₂ = Geometry.contravariant3(
+        getidx(A_space, loc, idx, hidx),
+        Geometry.LocalGeometry(face_space, idx, hidx),
+    )
+    return Geometry.Contravariant3Vector(Aⱼ₊₁₂ * 0.0)
```

Comment:
Why is this multiplying by zero?

Reply:
I want to set to 0, but if I set it to 0, I got errors

Reply:
Does A here represent the same thing as in the BB FCT? If so, in #888 we left only the value of A in the BCs. Why here we want this to be always zero?

Reply:
If it should be zero, then use `zero` (first determine the correct type), and don't call `contravariant3` (this performs operations). If it's not meant to be zero, then let's correct that issue

Reply:
I think both A and 0 are correct, A->third order upwind boundary condition implementation, 0->first order upwind boundary condition implementation. I tried both, 0-> gives better results in terms of overshoting

Reply:
You're saying that we should use 0 also for the BB FCT. I'd like these BCs to be consistent for the two operators, unless there is a specific reason why they shouldn't be. Otherwise it gets confusing for people who might be looking into this in the future and we can't remember why we made one choice over the other.

Reply:
I am not sure whether we should use 0 also for the BB FCT, could you try both for more tests to see which one is better? For Zalesak, using 0 reduces undershooting